dev/devicelab/lib/tasks/perf_tests.dart

```diff
@@ -1339,16 +1348,55 @@ class WebCompileTest {
     });
   }
 
-  /// Obtains the size and gzipped size of a file given by [fileName].
-  static Future<Map<String, int>> getSize(String fileName, {required String metric}) async {
+  /// Obtains the size and gzipped size of both [dartBundleFile] and [buildDir].
+  static Future<Map<String, int>> getSize({
+    /// Mapping of metric key name to file system path for directories to measure
+    Map<String, String> directories = const <String, String>{},
+    /// Mapping of metric key name to file system path for files to measure
+    Map<String, String> files = const <String, String>{},
+    required String metric,
+  }) async {
     final Map<String, int> sizeMetrics = <String, int>{};
 
-    final ProcessResult result = await Process.run('du', <String>['-k', fileName]);
-    sizeMetrics['${metric}_dart2js_size'] = _parseDu(result.stdout as String);
+    final Directory tempDir = Directory.systemTemp.createTempSync('perf_tests_gzips');
+    for (final MapEntry<String, String> entry in files.entries) {
+      final String key = entry.key;
+      final String filePath = entry.value;
+      ProcessResult result = await Process.run('du', <String>['-k', filePath]);
+      sizeMetrics['${metric}_${key}_size'] = _parseDu(result.stdout as String);
+
+      await Process.run('gzip',<String>['--keep', '-9', filePath]);
+      // gzip does not provide a CLI option to specify an output file, so
+      // instead just move the output file to the temp dir
+      final File compressedFile = File('$filePath.gz')
+          .renameSync(path.join(tempDir.absolute.path, '$key.gz'));
+      result = await Process.run('du', <String>['-k', compressedFile.path]);
+      sizeMetrics['${metric}_${key}_size_gzip'] = _parseDu(result.stdout as String);
+
+    }
 
-    await Process.run('gzip',<String>['-k', '9', fileName]);
-    final ProcessResult resultGzip = await Process.run('du', <String>['-k', '$fileName.gz']);
-    sizeMetrics['${metric}_dart2js_size_gzip'] = _parseDu(resultGzip.stdout as String);
+    for (final MapEntry<String, String> entry in directories.entries) {
+      final String key = entry.key;
+      final String dirPath = entry.value;
+      ProcessResult result = await Process.run('du', <String>['-k', dirPath]);
+
+      // when calling `du` on a directory, the last line is the sum of all its
+      // contents, thus the total size of the directory
+      final String lastLine = (result.stdout as String).trim().split('\n').last;
+      sizeMetrics['${metric}_${key}_size'] = _parseDu(lastLine);
+
+      // get size of compressed build directory
+      final String tarball = path.join(tempDir.absolute.path, '$key.tar.gz');
+      await Process.run('tar', <String>[
```

Comment:
If the `-c` trick works, I think you can do this directly with gzip and `-r` (recursive):

```
gzip --keep -9 -c -r dirPath > tarball
```

(That way we use the exact same compression level as the individual files above? I don't know what's the --gzip config for tar)

Reply:
explicitly pass `gzip -9` to `tar` now